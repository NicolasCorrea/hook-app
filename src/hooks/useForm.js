import { useState } from 'react'

export const useForm = (initialState = {}) => {
  const [formState, setFormState] = useState(initialState);
  const handleInputChange = ({ target }) => {
    setFormState({
      ...formState,
      [target.name]: target.value
    })
  }
  const reset = () => {
    setFormState(initialState);
  }
  
  return [formState, handleInputChange, reset];
}
