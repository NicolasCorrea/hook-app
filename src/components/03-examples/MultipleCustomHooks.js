import { useCounter } from "../../hooks/useCounter";
import "../02-useEffect/effects.css"
import { useFetch } from "../../hooks/useFetch"
export const MultipleCustomHooks = () => {

  const { counter, increment } = useCounter(1);
  const { loading, data } = useFetch(`https://www.breakingbadapi.com/api/quotes/${counter}`);
  const { author, quote } = !!data && data[0];
  return (
    <>
      <h1>Multiple Custom Hooks</h1>
      <hr />
      {
        loading ?
          <div className="alert alert-info" role="alert">
            <strong>Loading...</strong>
          </div>
          :
          <figure className="text-end">
            <blockquote className="blockquote text-right">
              <p className="mb-2">{quote}</p>
              <footer className="blockquote-footer">
                <cite title={author}>{author}</cite>
              </footer>
            </blockquote>
          </figure>
      }


      <button type="button" className="btn btn-primary" onClick={() => increment()}>Next Quote</button>


    </>
  )
}
