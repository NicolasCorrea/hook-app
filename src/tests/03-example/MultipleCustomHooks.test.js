import { shallow } from "enzyme"
import { MultipleCustomHooks } from "../../components/03-examples/MultipleCustomHooks"
import { useFetch } from "../../hooks/useFetch"
jest.mock("../../hooks/useFetch");
describe('test on <MultipleCustomHooks />', () => {
  test('should render', () => {
    useFetch.mockReturnValue({
      loading: true,
      data: null,
      error: null,
    })
    const wrapper = shallow(<MultipleCustomHooks />)
    expect(wrapper).toMatchSnapshot();
  })

  test('should show info', () => {
    useFetch.mockReturnValue({
      loading: false,
      data: [
        {
          author: "nicolas",
          quote: "Hello world"
        }
      ],
      error: null,
    })
    const wrapper = shallow(<MultipleCustomHooks />)

    expect(wrapper.find('.alert').exists()).toBe(false)
    expect(wrapper.find('.mb').exists()).toBe(false)
  })


})
