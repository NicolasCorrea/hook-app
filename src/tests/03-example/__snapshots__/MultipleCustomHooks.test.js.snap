// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test on <MultipleCustomHooks /> should render 1`] = `
<Fragment>
  <h1>
    Multiple Custom Hooks
  </h1>
  <hr />
  <div
    className="alert alert-info"
    role="alert"
  >
    <strong>
      Loading...
    </strong>
  </div>
  <button
    className="btn btn-primary"
    onClick={[Function]}
    type="button"
  >
    Next Quote
  </button>
</Fragment>
`;
